### ISSUE TITLE
SC2062 with deliberate shell expression for portable tab

### ISSUE THREAD

**Issue by peterjc** [koalaman/shellcheck#2948]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2062
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2062)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
thapbi_pict dump -d "$DB" | grep -v "^#" | grep -c Phytophthora$'\t'[a-z]
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
[Line 2:](javascript:setPosition(2, 52))
thapbi_pict dump -d "$DB" | grep -v "^#" | grep -c Phytophthora$'\t'[a-z]
                                                   ^-- [SC2062](https://www.shellcheck.net/wiki/SC2062) (warning): Quote the grep pattern so the shell won't interpret it.
```

#### Here's what I wanted or expected to see:

No warning: I want the shell to interpret the grep pattern and replace the $'\t' bit with a tab - and ugly but practical for a tab search on Linux (with gnu grep which has easier ways to do this) and macOS.

**Comment by peterjc** [comment]
Trying again this morning I solved this, my mistake:

```
#!/bin/bash
thapbi_pict dump -d "$DB" | grep -v "^#" | grep -c $'Phytophthora\t[a-z]'
```

Sorry for the noise.
